src/main/java/org/openrewrite/java/migrate/javax/AddColumnAnnotation.java

```diff
@@ -0,0 +1,82 @@
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.AddOrUpdateAnnotationAttribute;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.search.FindAnnotations;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+
+import java.util.Comparator;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddColumnAnnotation extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "ElementCollection annotations must be accompanied by a defined Column annotation";
+    }
+
+    @Override
+    public String getDescription() {
+        return "When an attribute is annotated with @ElementCollection, a separate table is created for the attribute that includes the attribute \n" +
+               "ID and value. In OpenJPA, the column for the annotated attribute is named element, whereas EclipseLink names the column based on \n" +
+               "the name of the attribute. To remain compatible with tables that were created with OpenJPA, add a @Column annotation with the name \n" +
+               "attribute set to element.";
+    }
+
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesType<>("javax.persistence.ElementCollection", true),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext ctx) {
+                        // Exit if var does not have @ElementCollection
+                        if (FindAnnotations.find(multiVariable, "@javax.persistence.ElementCollection").isEmpty()) {
+                            return multiVariable;
+                        }
+
+                        // Create and add @Column annotation
+                        if (FindAnnotations.find(multiVariable, "@javax.persistence.Column").isEmpty()) {
+                            maybeAddImport("javax.persistence.Column");
+                            return JavaTemplate.builder("@Column(name = \"element\")")
+                                    .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "javax.persistence-api-2.2"))
+                                    .imports("javax.persistence.Column")
+                                    .build()
+                                    .apply(getCursor(), multiVariable.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName)));
+                        }
+
+                        // Update existing @Column annotation
+                        doAfterVisit(new AddOrUpdateAnnotationAttribute(
```

Comment:
Since `doAfterVisit` applies to the whole file, this may unintentionally change an existing `@Column` annotation on a varDecl that does not have an `@ElementCollection` annotation.

Reply:
A way around here might be to directly call visit methods on the `.getVisitor()` returned; haven't tried that specifically here yet, but could be a way around.